### ISSUE TITLE
[a11y]: ComposedModal 2.4.1: Content is not within a landmark element

### ISSUE THREAD

**Issue by gcattan** [carbon-design-system/carbon#11034]
### Package

carbon-components-react

### Browser

Chrome

### Operating System

Windows

### Package version

v7.55.1

### Automated testing tool and ruleset

IBM Accessibility Tester

### Description

Content is not within a landmark element.

### WCAG 2.1 Violation

2.4.1 Bypass Block

### CodeSandbox example

https://j0bubn.csb.app/

### Steps to reproduce

Create a ComposedModal (see sandbox), and scan with IBM Accessibility checker:

![image](https://user-images.githubusercontent.com/6229031/159478845-fa1fbeda-d233-4b0f-b31a-44c554a8bf6e.png)

![image](https://user-images.githubusercontent.com/6229031/159478918-3a988d44-07d6-4792-84d6-a4f8688770a5.png)

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems
